### PR TITLE
[KYUUBI #3672][Subtask] Get table owner for DDL/DML v1 commands

### PR DESCRIPTION
**Pull request by zhouyifan279**
### _Why are the changes needed?_
Subtask of #3607 

### _How was this patch tested?_
- [x] Add some test cases that check the changes thoroughly including negative and positive cases if possible

- [ ] Add screenshots for manual tests if appropriate

- [x] [Run test](https://kyuubi.apache.org/docs/latest/develop_tools/testing.html#running-tests) locally before make a pull request

### CODE REVIEW

**Comment by zhouyifan279** [comment]
cc @bowenliang123 @pan3793

**Review by pan3793 — APPROVED** [review]
LGTM, only minor things

**Review by bowenliang123 — APPROVED** [review]

**Comment by bowenliang123** [comment]
LGTM

File: extensions/spark/kyuubi-spark-authz/src/test/scala/org/apache/kyuubi/plugin/spark/authz/PrivilegesBuilderSuite.scala [hunk 1]
```diff
@@ -123,24 +123,34 @@ abstract class PrivilegesBuilderSuite extends AnyFunSuite
...
+        // toLowerCase because of: SPARK-38587
+        val plan =
+          sql(s"ALTER TABLE ${reusedDb.toLowerCase}.$oldTableShort" +
+            s" RENAME TO $t").queryExecution.analyzed
+        // re-create oldTable because we needs to get table owner from table metadata later
+        sql(create)
+
+        val operationType = OperationType(plan.nodeName)
+        assert(operationType === ALTERTABLE_RENAME)
+        val tuple = PrivilegesBuilder.build(plan, spark)
```
[the remark this answers is not in the document]
Comment (zhouyifan279) [reply extensions/spark/kyuubi-spark-authz/src/test/scala/org/apache/kyuubi/plugin/spark/authz/PrivilegesBuilderSuite.scala:140]: Quite a lot places need to change. I suggest to import them in a separate PR.

File: extensions/spark/kyuubi-spark-authz/src/test/scala/org/apache/kyuubi/plugin/spark/authz/PrivilegesBuilderSuite.scala [hunk 2]
```diff
@@ -332,8 +348,11 @@ abstract class PrivilegesBuilderSuite extends AnyFunSuite
     assert(po0.privilegeObjectType === PrivilegeObjectType.TABLE_OR_VIEW)
     assert(po0.dbname equalsIgnoreCase reusedDb)
     assert(po0.objectName equalsIgnoreCase reusedPartTableShort)
-    // ignore this check as it behaves differently across spark versions
-    // assert(po0.columns === Seq("key", "value", "pid"))
+    if (isSparkV32OrGreater) {
+      // ignore this check as it behaves differently across spark versions
```
[the remark this answers is not in the document]
Comment (zhouyifan279) [reply extensions/spark/kyuubi-spark-authz/src/test/scala/org/apache/kyuubi/plugin/spark/authz/PrivilegesBuilderSuite.scala]: Done

File: extensions/spark/kyuubi-spark-authz/src/test/scala/org/apache/kyuubi/plugin/spark/authz/PrivilegesBuilderSuite.scala (hunk 1 above)
[the remark this answers is not in the document]
Comment (pan3793) [reply extensions/spark/kyuubi-spark-authz/src/test/scala/org/apache/kyuubi/plugin/spark/authz/PrivilegesBuilderSuite.scala:140]: it's fine

**Comment by pan3793** [comment]
Thanks, merging to master

**Pull request closed without merging**